sdk/turing/ensembler.py

```diff
@@ -55,20 +55,27 @@ def initialize(self, artifacts: dict):
         """
         pass
 
-    def predict(self, context, model_input: pandas.DataFrame) -> \
-            Union[numpy.ndarray, pandas.Series, pandas.DataFrame]:
-        prediction_columns = PyFunc._get_columns_with_prefix(model_input, PyFunc.PREDICTION_COLUMN_PREFIX)
-        treatment_config_columns = PyFunc._get_columns_with_prefix(model_input, PyFunc.TREATMENT_CONFIG_COLUMN_PREFIX)
-
-        return model_input \
-            .rename(columns=prediction_columns) \
-            .rename(columns=treatment_config_columns) \
-            .apply(lambda row:
-                   self.ensemble(
-                       features=row.drop(prediction_columns.values()).drop(treatment_config_columns.values()),
-                       predictions=row[prediction_columns.values()],
-                       treatment_config=row[treatment_config_columns.values()]
-                   ), axis=1, result_type='expand')
+    def predict(self, context, model_input: Union[pandas.DataFrame, Dict[str, Any]]) -> \
```

Comment:
Can we split this method into two private methods with strict signature types? I.e.
```
def _ensemble_batch(self, model_input: pandas.DataFrame) -> ...

def _ensemble_service(self, model_input: Dict[str, Any]) -> ...
```
And then call these methods from inside the `predict` methods, based on the actual type of the `model_input`

Reply:
Ok that's done :)